File: app/src/main/java/me/devsaki/hentoid/activities/sources/JapBeautiesActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class JapBeautiesActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "japanesebeauties.one";
    private static final String[] DIRTY_ELEMENTS = {".unit-main", ".unit-dt-blk", ".unit-mobblk"};
    private static final String[] GALLERY_FILTER = {"japanesebeauties.one/.*/.*/.*/.*"};

    Site getStartSite() {
        return Site.JAPBEAUTIES;
    }

    @Override
    boolean allowMixedContent() {
        return false;
    }


    @Override
    protected CustomWebViewClient createWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(getStartSite(), GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        client.addRemovableElements(DIRTY_ELEMENTS);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/JapBeautiesActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

private const val DOMAIN_FILTER = "japanesebeauties.one"
private val GALLERY_FILTER =
    arrayOf("japanesebeauties.one/[\\w_%\\-]+/[\\w_%\\-]+/[\\w_%\\-]+/([\\w_%\\-]+){0,1}$")
private val DIRTY_ELEMENTS = arrayOf(".unit-main", ".unit-dt-blk", ".unit-mobblk")

class JapBeautiesActivityK : BaseBrowserActivity() {
    override fun getStartSite(): Site {
        return Site.JAPBEAUTIES
    }

    override fun allowMixedContent(): Boolean {
        return false
    }


    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        client.addRemovableElements(*DIRTY_ELEMENTS)
        return client
    }
}